doc/changes/v3.0.0.rst

```diff
@@ -0,0 +1,356 @@
+
+Significant updates for the GPU-based external memory training implementation. (:pr:`10924`,
+:pr:`10895`, :pr:`10766`, :pr:`10544`, :pr:`10677`, :pr:`10615`, :pr:`10927`, :pr:`10608`, :pr:`10711`)
+
+- GPU-based external memory supports both batch-based and sampling-based training. Before
+  the 3.0 release, XGBoost concatenates the data during training and stores the cache on
+  disk. In 3.0, XGBoost can now stage the data on the host and fetch them by
+  batch. (:pr:`10602`, :pr:`10595`, :pr:`10606`, :pr:`10549`, :pr:`10488`, :pr:`10766`, :pr:`10765`, :pr:`10764`, :pr:`10760`, :pr:`10753`,
+  :pr:`10734`, :pr:`10691`, :pr:`10713`, :pr:`10826`, :pr:`10811`, :pr:`10810`, :pr:`10736`, :pr:`10538`)
+- XGBoost can now utilize `NVLink-C2C` for GPU-based external memory training and can
+  handle up to terabytes of data.
+- Support prediction cache (:pr:`10707`).
+- Automatic page concatenation for improved GPU utilization (:pr:`10887`).
+- Improved quantile sketching algorithm for batch-based inputs. See section for new
+  :ref:`features <3_0_features>` for more info.
+- Optimization for nearly-dense input, see the section for :ref:`optimization
+  <3_0_optimization>` for more info.
+
+See our latest document for details :doc:`/tutorials/external_memory`. The PyPI package
+(``pip install``) doesn't have ``RMM`` support, which is required by the GPU external
+memory implementation. To experiment, you can compile XGBoost from source or wait for the
+conda package to be available.
+
+.. _3_0_networking:
+
+**********
+Networking
+**********
+
+Continuing the work from the previous release, we updated the network module to improve
+reliability. (:pr:`10453`, :pr:`10756`, :pr:`11111`, :pr:`10914`, :pr:`10828`, :pr:`10735`, :pr:`10693`, :pr:`10676`, :pr:`10349`,
+:pr:`10397`, :pr:`10566`, :pr:`10526`, :pr:`10349`)
+
+The timeout option is now supported for NCCL using the NCCL asynchronous mode (:pr:`10850`,
+:pr:`10934`, :pr:`10945`, :pr:`10930`).
+
+In addition, a new :py:class:`~xgboost.collective.Config` class is added for users to
+specify various options including timeout, tracker port, etc for distributed
+training. Both the Dask interface and the PySpark interface support the new
+configuration. (:pr:`11003`, :pr:`10281`, :pr:`10983`, :pr:`10973`)
+
+****
+SYCL
+****
+
+Continuing the work on the SYCL integration, there are significant improvements in the
+feature coverage for this release from more training parameters and more objectives to
+distributed training, along with various optimization (:pr:`10884`, :pr:`10883`). Newly introduced
+features include:
+
+- Dask support for distributed training (:pr:`10812`)
+
+- Various training procedures, including split evaluation (:pr:`10605`, :pr:`10636`), grow policy
+  (:pr:`10690`, :pr:`10681`), cached prediction (:pr:`10701`).
+
+- Updates for objective functions. (:pr:`11029`, :pr:`10931`, :pr:`11016`, :pr:`10993`, :pr:`11064`, :pr:`10325`)
+
+- Support f32 devices (:pr:`10702`)
```

Comment:
Since fp32 support is still broken, I suggest to replace it by "Ongoing work on fp32 devices support"

Reply:
Done. Although I think a software implementation of f64 makes more sense than changing existing libraries/applications to use only f32.